app/models/topic_tracking_state.rb

```diff
@@ -71,6 +72,15 @@ def self.unread_channel_key(user_id)
     "/unread/#{user_id}"
   end
 
+  def self.publish_muted(post)
+    user_ids = post.topic.topic_users.where(notification_level: NotificationLevels.all[:muted]).order(notifications_changed_at: :desc).limit(100).pluck(:user_id)
```

Comment:
I prefer to look at last user activity ... join the `users` table looking at `last_seen_at`

When you muted matters a lot less vs when you were last around. 

In fact I would filter out all users that have not been seen in the last 7 days as well.